openedx/core/djangoapps/course_groups/views.py

```diff
@@ -29,76 +31,119 @@ def json_http_response(data):
-    ]
 
-    return json_http_response({'success': True,
-                               'cohorts': all_cohorts})
+def _get_cohort_representation(cohort, course):
+    """
+    Returns a JSON representation of a cohort.
+    """
+    group_id, partition_id = cohorts.get_group_info_for_cohort(cohort)
+    return {
+        'name': cohort.name,
+        'id': cohort.id,
+        'user_count': cohort.users.count(),
+        'assignment_type': cohorts.CohortAssignmentType.get(cohort, course),
+        'user_partition_id': partition_id,
+        'group_id': group_id
+    }
 
 
+@require_http_methods(("GET", "PUT", "POST", "PATCH"))
 @ensure_csrf_cookie
-@require_POST
-def add_cohort(request, course_key_string):
+@expect_json
+@login_required
+def cohort_handler(request, course_key_string, cohort_id=None):
     """
-    Return json of dict:
-    {'success': True,
-     'cohort': {'id': id,
-                'name': name}}
-
-                or
-
-    {'success': False,
-     'msg': error_msg} if there's an error
+    The restful handler for cohort requests. Requires JSON.
+    GET
+        If a cohort ID is specified, returns a JSON representation of the cohort
+            (name, id, user_count, assignment_type, user_partition_id, group_id).
+        If no cohort ID is specified, returns the JSON representation of all cohorts.
+           This is returned as a dict with the list of cohort information stored under the
+           key `cohorts`.
+    PUT or POST or PATCH
+        If a cohort ID is specified, updates the cohort with the specified ID. Currently the only
+        properties that can be updated are `name`, `user_partition_id` and `group_id`.
+        Returns the JSON representation of the updated cohort.
+        If no cohort ID is specified, creates a new cohort and returns the JSON representation of the updated
+        cohort.
     """
-    # this is a string when we get it here
     course_key = SlashSeparatedCourseKey.from_deprecated_string(course_key_string)
-
-    get_course_with_access(request.user, 'staff', course_key)
-
-    name = request.POST.get("name")
-    if not name:
-        return json_http_response({'success': False,
-                                   'msg': "No name specified"})
-
-    try:
-        cohort = cohorts.add_cohort(course_key, name)
-    except ValueError as err:
-        return json_http_response({'success': False,
-                                   'msg': str(err)})
-
-    return json_http_response({
-        'success': 'True',
-        'cohort': {
-            'id': cohort.id,
-            'name': cohort.name
-        }
-    })
+    course = get_course_with_access(request.user, 'staff', course_key)
+    if request.method == 'GET':
+        if not cohort_id:
+            all_cohorts = [
+                _get_cohort_representation(c, course)
+                for c in cohorts.get_course_cohorts(course)
+            ]
+            # TODO: change to just directly returning the lists.
```

Comment:
I dug around and couldn't find any consistency. The API documentation suggests that we should consider making this a paginated interface, but doesn't say what that looks like:

https://openedx.atlassian.net/wiki/display/AC/API+Thoughts

I think you should probably leave it as is for now, and in the future we should start to use the Django REST Framework for all of our APIs. For an example, see how DedX is using it in the user_api:

https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/user_api/views.py#L168